src/main/java/io/jenkins/pluginhealth/scoring/probes/LastCommitDateProbe.java

```diff
@@ -0,0 +1,77 @@
+package io.jenkins.pluginhealth.scoring.probes;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.ZonedDateTime;
+
+import io.jenkins.pluginhealth.scoring.model.Plugin;
+import io.jenkins.pluginhealth.scoring.model.ProbeResult;
+import io.jenkins.pluginhealth.scoring.model.ResultStatus;
+
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.lib.ObjectId;
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.eclipse.jgit.revwalk.RevWalk;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+/**
+ * Using the analysis done by {@link SCMLinkValidationProbe#}, this probe determines the last commit date of the plugin's repository.
+ */
+@Component
+@Order(value = LastCommitDateProbe.ORDER)
+public class LastCommitDateProbe extends Probe {
+    private static final Logger LOGGER = LoggerFactory.getLogger(LastCommitDateProbe.class);
+
+    public static final int ORDER = SCMLinkValidationProbe.ORDER + 1;
+
+    @Override
+    public ProbeResult doApply(Plugin plugin) {
+        try {
+            if (plugin.getDetails().get(SCMLinkValidationProbe.KEY) == null) {
+                LOGGER.error("Couldn't run {} on {} because previous SCMLinkValidationProbe has null value in database", key(), plugin.getName());
+                return ProbeResult.error(key(), "SCM link has not been probed yet");
+            }
+
+            if (plugin.getDetails().get(SCMLinkValidationProbe.KEY).status() == ResultStatus.SUCCESS) {
+                try {
+                    final Path tempDirectory = Files.createTempDirectory(plugin.getName());
+                    try (Git git = Git.cloneRepository().setURI(plugin.getScm()).setDirectory(tempDirectory.toFile()).call()) {
+                        final RevCommit head = new RevWalk(git.getRepository()).parseCommit(ObjectId.fromString("HEAD"));
+                        final ZonedDateTime zonedDateTime = ZonedDateTime.ofInstant(
+                            head.getAuthorIdent().getWhenAsInstant(),
+                            head.getAuthorIdent().getZoneId()
+                        );
+                        return ProbeResult.success(key(), zonedDateTime.toString());
+                    } finally {
+                        Files.delete(tempDirectory);
```

Comment:
Please correct me if I'm wrong, but I'm observing that this method does not delete the plugin directory (and all of its files and subdirectories)  and I'm hitting this exception: `java.nio.file.DirectoryNotEmptyException: /tmp/job-log-logger-plugin6397033103289588371` for example. Due to this, the status of this probe results for each plugin in the DB is a failure. Do you see this as well?

Reply:
More than just that, it is not able to get the object id `HEAD`, so I'm working on it.

Reply:
It works if we pass `git.getRepository().resolve(Constants.HEAD)` to the `parseCommit()`, but I'm not sure if this is the ideal way to do it.

Reply:
this is how I did it.

Reply:
Just saw your commits :+1: